### PR TITLE
(doc) Create complex_routing.rs

### PR DESCRIPTION
**Pull request by bspeice**
Add a slightly more nuanced and realistic routing example.
I'm not sure if this should be committed to the router repository, as it technically depends on code from there.
Intended to address issue #311

### CODE REVIEW

**Comment by untitaker** [comment]
Thanks for this PR, however I don't understand, why is this more realistic?

**Comment by bspeice** [comment]
I couldn't find any examples on how to route to things that weren't static functions. This adds the ability to encapsulate other state that can be used in responses

File: examples/complex_routing.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+use iron::Handler;
+use iron::status;
+use iron::IronResult;
+use iron::Response;
+use iron::Request;
+use iron::Iron;
+use router::Router;
+
+struct EchoHandler {
```
Comment (untitaker) [inline examples/complex_routing.rs]: This doesn't "echo" anything the client sent.

**Comment by untitaker** [comment]
Ah, the import of routing fails, I think this PR should be filed against the routing repo since the crate is available there.

**Comment by bspeice** [comment]
OK. It shouldn't be hard to do the same example in both cases, I'll get the pull requests whipped up later today.

**Comment by bspeice** [comment]
Closing, as there's already an example of struct-based responses in the routing example.

**Comment by untitaker** [comment]
You mean your PR?

**Comment by bspeice** [comment]
No, the actual [examples/simple_routing.rs](https://github.com/iron/iron/blob/master/examples/simple_routing.rs) uses a struct-based handler. I don't know if this would qualify as redundant.
It's pretty trivial to remove the routing dependency, I can go ahead and clean it up here if you want.

**Comment by drusellers** [comment]
Although this has been closed, this is a huge need. Iron has great support for static functions, but the next step is how do I get my, for lack of a better word, `controller` vibe going on?

**Comment by untitaker** [comment]
There's an example in the routing repo.

**Pull request closed without merging**